src/pyscaffold/templates/tox_ini.template

```diff
@@ -20,6 +20,17 @@ commands =
     pytest {posargs}
 
 
+[testenv:lint]
```

Comment:
Maybe this testenv should not be added by default to every user, considering that `pre-commit` will not be installed by everyone (so the pre-commit hooks file will be absent)...

Reply:
Hm... and if we let `github-actions` require `pre-commit`? Now I understand the comment before. Since `pre-commit` becomes so important, we could also in a 5.0 release switch to making it as default and providing a `no-pre-commit` flag instead. 

For the time being, does it hurt to have a `testenv:lint` that is never called if the user doesn't run it? In the `github_ci_workflow.template` one could add or not add the `lint`-step depending if `--precommit` was chosen for the time being. Or just adding `--precommit` in case `--github-actions` was chosen. I would argue that one always wants some linting checks when working with CI/CD.

Reply:
I agree with adding `pre-commit` by default in v5, however we have to discuss how *exactly* we are doing that[^1]

[^1]: Before v4, running `putup --pre-commit` had only the side effect of producing some extra file. Therefore, there was no problem in having it as a default (if the user didn't want to run `pre-commit --install` no changes in workflow would happen). Now, however, `putup --pre-commit` will try to automatically run `pre-commit install`, so leaving it as a default, without changing anything else will change the dev's workflow.

> does it hurt to have a `testenv:lint` that is never called if the user doesn't run it?

Currently there will be no problem for users running `--github-actions`, because this extension will also automatically activate `--pre-commit` even if `--pre-commit` is not passed. The situation without `--github-actions` is the one that changes.

What might happen if the default `tox.ini` template have a `testenv:lint` is:
- an user that runs `putup` without `--pre-commit`
- the user runs `tox -av` to find out which tasks are available
- the user sees `lint` there
- the user runs `tox -e lint`
- tox fails with the following error message:
  ```
  An error has occurred: InvalidConfigError:
  =====> .pre-commit-config.yaml is not a file
  Check the log at ~/.cache/pre-commit/pre-commit.log
  ```

Reply:
In https://github.com/pyscaffold/pyscaffold/pull/493#issuecomment-959797846, https://github.com/pyscaffold/pyscaffold/pull/493#issuecomment-961381764 and https://github.com/pyscaffold/pyscaffold/pull/493#issuecomment-961385956 I was playing around with the idea of having an API were we can comment/uncomment lines of the files in the tree structure[^1]...

[^1]: While I agree that enabling general modifications to files is very risky and error-prone, I think the capability of commenting/uncommenting and adding/removing entire blocks of text is less problematic...

Using the same line of thought, one solution here would be having the `testenv:lint` commented out by default and then the `--pre-commit` extension could uncomment it (or simply adding it in the case it is not there in the form of a commented block).

Other viable solutions is to make the default `tox.ini` template "aware" of pre-commit. So when the `--pre-commit` extension is activated, the testenv is added to the tox template. (The downside here is increasing coupling)

Reply:
I think your "commenting and uncommenting"-solution is the best in this case. We should find a nice and robust way to do this.